mypyc/test/test_serialization.py

```diff
@@ -58,7 +58,7 @@ def assert_blobs_same(x: Any, y: Any, trail: Tuple[Any, ...]) -> None:
         assert x.keys() == y.keys(), "Keys mismatch at {}".format(trail)
         for k in x.keys():
             assert_blobs_same(x[k], y[k], trail + (k,))
-    elif isinstance(x, Iterable) and not isinstance(x, str):
+    elif isinstance(x, Iterable) and not isinstance(x, (str, set)):
```

Comment:
It is a bit odd that we treat sets different from other iterables (I understand treating strs differently)

Reply:
Sets have unpredictable ordering, so we can't use the special casing for them.